src/Lecture2.hs

```diff
@@ -191,7 +227,13 @@ verify that.
 [1,2,3,4,7]
 -}
 merge :: [Int] -> [Int] -> [Int]
-merge = error "TODO"
+merge [] [] = []
+merge [] ys = ys
+merge xs [] = xs
+merge [x] [y] = if x < y then [x, y] else [y, x]
+merge (x:xs) [y] = if y < x then [y,x] ++ xs else [x] ++ merge xs [y]
+merge [x] (y:ys) = if x < y then [x,y] ++ ys else [y] ++ merge [x] ys
+merge xs (y:ys) = merge (merge xs [y]) ys
```

Comment:
This looks like lots of redundant cases 🧹 

Try pattern matching on the following pattern and going from there:

```haskell
merge (x : xs) (y : ys)
```